### ISSUE TITLE
How do JSON result bodies become Swagger model documentation?

### ISSUE THREAD

**Issue by refried** [http4s/rho#62]
In the examples, I see the tag trait `AutoSerializable` and the encoder `implicit def jsonWritable[A <: AutoSerializable]: EntityEncoder[A]`, which uses jackson's `write` to produce JSON, but that isn't part of the process for generating the Swagger model, is it?

**Comment by refried** [comment]
P.S. I am looking at the Swagger 2.0 `definitions:` and getting excited because they match what I want to start outputting.

**Comment by bryce-anderson** [comment]
No, the ```AutoSerializable``` trait isn't part of the swagger process. It is an experimental construct I made up to allow you to easily send case classes. I don't think it will ever escape the examples as it feels a little 'magic'. If it belongs anywhere permanently, it would be in ```http4s``` proper.

**Comment by refried** [comment]
No, ok.  So `AutoSerializable` is just part of the way the example project produces the response body.

Then how does the example project's json become a swagger json definition?

**Comment by refried** [comment]
Basically, I want to understand how to start extending this capability.  Complicating things a little bit is the fact that Swagger 2.0 is much more expressive, and I kind of want to jump onto that train, although it is in motion.

**Comment by bryce-anderson** [comment]
The ```SwaggerSupport``` trait is tasked with maintaining and rendering the documentation [here](https://github.com/http4s/rho/blob/master/swagger/src/main/scala/org/http4s/rho/swagger/SwaggerSupport.scala#L43) in master and [here](https://github.com/http4s/rho/blob/topic/swagger2.0/swagger/src/main/scala/org/http4s/rho/swagger/SwaggerSupport.scala#L39) in the swagger2.0 branch. @cencarnacion has submitted an excellent PR concerning swagger 2.0 which I hope to merge in the very near future, so its probably worth looking at that code from now on.

**Comment by refried** [comment]
Ok, I see it, and in particular I'm looking at `org.http4s.rho.swagger.TypeBuilder` currently.
